fix docs and enable CI on `main` branch

hds opened the pull request:
## Motivation

The new `main` branch is forked from the `v0.1.x` branch. It will be
made the default branch and going forward we will merge PRs to this
branch first (and then forward port to a new `v0.2.x` branch forked from
`master`).

It looks like Netlify jobs weren't running on the `v0.1.x` branch, and so
there were quite a few errors in the docs on that branch (which isn't great
because those are the ones that get published to docs.rs).

## Solution

Separate to this PR, we've enabled Netlify on the `main` branch, and this
change fixes all the errors that were present in the docs.

This change sets the GitHub actions to run on the `main` branch instead
of `v0.1.x`. It also adds some text in the root README.md which
describes the branch set-up.

Refs: #3294

kaffarell commented:
Not so important, but maybe we can also change the links in the README? So changing the badge from master to main and maybe remove the "Documentation (master branch)" link as it's broken anyway.

hds commented:
@kaffarell Good call. I've replaced references to assets in `master` with `main` instead. I removed the text master docs link from the main README, but kept all the others, just renamed them from `master` to `v0.2.x` and fixed the links (they should go to tracing.rs instead of tracing-rs.netlify.com).

djc commented:
@hds thanks for doing the legwork on this!
